src/ioctls/device.rs

```diff
@@ -0,0 +1,97 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::fs::File;
+use std::io;
+use std::os::unix::io::{AsRawFd, FromRawFd, RawFd};
+
+use kvm_bindings::kvm_device_attr;
+
+use ioctls::Result;
+use kvm_ioctls::KVM_SET_DEVICE_ATTR;
+use sys_ioctl::ioctl_with_ref;
+
+/// Wrapper over the file descriptor obtained when creating an emulated device in the kernel.
+pub struct DeviceFd {
```

Comment:
The structure need to be publicly exported in src/lib.rs to be accessible outside of the crate.

Reply:
Done, see lib.rs file.